bpf/lib/ratelimit.h

```diff
@@ -24,8 +32,25 @@ struct {
 	__uint(max_entries, 1024);
 } RATELIMIT_MAP __section_maps_btf;
 
+struct ratelimit_metrics_key {
+	__u32 usage;
+};
+
+struct ratelimit_metrics_value {
+	__u64 dropped;
+};
+
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__type(key, struct ratelimit_metrics_key);
+	__type(value, struct ratelimit_metrics_value);
+	__uint(pinning, LIBBPF_PIN_BY_NAME);
+	__uint(max_entries, 64);
+	__uint(map_flags, CONDITIONAL_PREALLOC);
+} RATELIMIT_METRICS_MAP __section_maps_btf;
```

Comment:
Why can't this use the existing metrics map instead?

Reply:
Because the existing metrics map and this ratelimit metrics map have two completely different purposes. Existing map is not a generic map that exports any type of metric from bpf.

Metrics map measures stats about packets flowing through bpf code, in particular overall count and bytes. This new map measures stats about bpf rate limiter. Rate limiting stats might be completely unrelated to packets (they are related for ICMPv6, but not for events). Current metrics map doesn't track any information related to events exported to events map.

One could think of some neat way of introducing unions to existing metrics map key / value structs, so that it can cover the existing metrics, these new metrics and more future metrics as well. It would risk breaking current metrics, be less readable, it wouldn't be pretty and I don't see why would it be worth it. Unless there is an actual reason to merge these maps, I would rather stay with the current approach.